### PR TITLE
[closes #49] mst prims algorithm

### LINKED ISSUE #49: Implement Prim or Kruskal's Algorithm for finding the miminum spanning tree of the graph.
Depends on #42 
Subtask for #14

**Definition of Done**
1. Prim or Kruskal's algorithm should be implemented for use with a Graph object.
2. Must find the minimum spanning tree visiting all stadiums.
3. Unit tests for total distance and path taken must be written and passing.
4. Code must be fully documented using Doxygen comment blocks.

**Test Scenarios**
1. Verify total distance of a path is accurate.
2. Verify path is accurate.
3. Verify that the original Graph is not modified in any way by the algorithm.

### PR DESCRIPTION
**Pull request by EDKarlsson**
Shortest path is 7060, test suite will run every possible starting point till it finds the shortest path.

### CODE REVIEW

**Comment by EDKarlsson** [comment]
Oh need to use the `fixed-backup.txt` to fix the `data.db` file in develop.

File: src/header/vertex.h [hunk 1]
```diff
@@ -260,6 +297,21 @@ class Vertex
         }
     }
```
Comment (ozyx) [inline src/header/vertex.h:304]: woops

File: src/header/vertexqueue.h [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+     */
+    Vertex removeMin()
+    {
+        Vertex rootNode;
+        rootNode = this->root();
+        vertexMap.remove(rootNode.getName());
+        this->remove(1);
+        return rootNode;
+    }
+
```
Comment (ozyx) [inline src/header/vertexqueue.h:150]: need doxygen blocks here on down

File: test/test_main.cpp [hunk 3]
```diff
@@ -16,12 +17,16 @@ private slots:
```
Comment (ozyx) [inline test/test_main.cpp:17]: comments

**Comment by ozyx** [comment]
good 💩

**Pull request merged**